splash/qtutils.py

```diff
@@ -141,9 +140,19 @@ def qurl2ascii(url):
     return url
 
 
+def to_qurl(s):
+    if isinstance(s, QUrl):
+        return s
+
+    if isinstance(s, unicode):
+        s = s.encode('utf-8')
+
+    return QUrl.fromEncoded(s)
```

Comment:
Not sure, my guess is documentation error. I tested different ways of
converting to a QUrl and this is the only one that did the right thing
every time.
On Oct 13, 2015 6:27 PM, "Mikhail Korobov" <notifications@github.com> wrote:

> In splash/qtutils.py
> <https://github.com/scrapinghub/splash/pull/314#discussion_r41854408>:
>
> > @@ -141,9 +140,19 @@ def qurl2ascii(url):
> >      return url
> >
> >
> > +def to_qurl(s):
> > +    if isinstance(s, QUrl):
> > +        return s
> > +
> > +    if isinstance(s, unicode):
> > +        s = s.encode('utf-8')
> > +
> > +    return QUrl.fromEncoded(s)
>
> Docs <http://doc.qt.io/qt-5/qurl.html#fromEncoded> say "Parses input and
> returns the corresponding QUrl. input is assumed to be in encoded form,
> containing only ASCII characters.", but we're not escaping non-ASCII chars.
> Is it an issue with Qt documentation, or a feature they don't officially
> support?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/scrapinghub/splash/pull/314/files#r41854408>.
>